src/lightning_fabric/README.md

```diff
@@ -1 +1,167 @@
-# TODO
+<div align="center">
+
+<img src="https://pl-public-data.s3.amazonaws.com/assets_lightning/fabric_logo.png" width="400px">
+
+**Fabric is the fast and lightweight way to scale PyTorch models without boilerplate**
+
+______________________________________________________________________
+
+<p align="center">
+  <a href="https://lightning.ai/">Website</a> •
+  <a href="https://lightning.ai/docs/fabric/">Docs</a> •
+  <a href="#getting-started">Getting started</a> •
+  <a href="#faq">FAQ</a> •
+  <a href="#asking-for-help">Help</a> •
+  <a href="https://www.pytorchlightning.ai/community">Slack</a>
+</p>
+
+</div>
+
+## Maximum flexibility, minimum code changes
+
+With just a few code changes, run any PyTorch model on any distributed hardware, no boilerplate!
+
+- Easily switch from running on CPU to GPU (Apple Silicon, CUDA, …), TPU, multi-GPU or even multi-node training
+- Use state-of-the-art distributed training strategies (DDP, FSDP, DeepSpeed) and mixed precision out of the box
+- All the device logic boilerplate is handled for you
+- Designed with multi-billion parameter models in mind
+- Build your own custom Trainer using Fabric primitives for training checkpointing, logging, and more
+
+```diff
++ import lightning as L
+
+  import torch
+  import torch.nn as nn
+  from torch.utils.data import DataLoader, Dataset
+
+  class PyTorchModel(nn.Module):
+      ...
+
+  class PyTorchDataset(Dataset):
+      ...
+
++ fabric = L.Fabric(accelerator="cuda", devices=8, strategy="ddp")
++ fabric.launch()
+
+- device = "cuda" if torch.cuda.is_available() else "cpu
+  model = PyTorchModel(...)
+  optimizer = torch.optim.SGD(model.parameters())
++ model, optimizer = fabric.setup(model, optimizer)
+  dataloader = DataLoader(PyTorchDataset(...), ...)
++ dataloader = fabric.setup_dataloaders(dataloader)
+  model.train()
+
+  for epoch in range(num_epochs):
+      for batch in dataloader:
+          input, target = batch
+-         input, target = input.to(device), target.to(device)
+          optimizer.zero_grad()
+          output = model(input)
+          loss = loss_fn(output, target)
+-         loss.backward()
++         fabric.backward(loss)
+          optimizer.step()
+          lr_scheduler.step()
+```
+
+______________________________________________________________________
+
+# Getting started
+
+## Install Lightning
+
+<details>
+
+<summary>Prerequisites</summary>
+
+> TIP: We strongly recommend creating a virtual environment first.
+> Don’t know what this is? Follow our [beginner guide here](https://lightning.ai/docs/stable/install/installation.html).
+
+- Python 3.8.x or later (3.8.x, 3.9.x, 3.10.x, ...)
+
+</details>
+
+```bash
+pip install -U lightning
+```
+
+## Convert your PyTorch to Fabric
+
+1. Create the `Fabric` object at the beginning of your training code.
+
+```
```

Comment:
Nice!
If you indent the code blocks by 4 spaces, aligning it with the start of the bullet point "Create the Fabric...", the code blocks will be better aligned with the bullet points in the rendered output :)